### PR TITLE
Pruning storer fixes

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed concurrent persisters opening
- fixed a leaking pointer on an opened persister

### CODE REVIEW

File: storage/pruning/pruningStorer.go [hunk 1]
```diff
@@ -765,10 +766,12 @@ func (ps *PruningStorer) changeEpochWithExisting(epoch uint32) error {
...
 		if p.getIsClosed() {
-			_, err = ps.persisterFactory.Create(p.path)
-			if err != nil {
-				return err
+			db, errCreate := ps.persisterFactory.Create(p.path)
+			if errCreate != nil {
+				return errCreate
 			}
+
+			p.setPersisterAndIsClosed(db, false)
```
Comment (AdoAdoAdo) [inline storage/pruning/pruningStorer.go:774]: :+1:

**Review by AdoAdoAdo — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]
good catch :+1:

File: storage/pruning/pruningStorer.go (hunk 1 above)
Comment (SebastianMarian) [reply storage/pruning/pruningStorer.go:774]: The method name is confusing

**Review by SebastianMarian — APPROVED** [review]

**Comment by iulianpascalau** [comment]
> good catch +1

:+1: good catch

**Review by gabi-vuls — APPROVED** [review]
System test passed.
image

**Pull request merged**